src/jenkins/BuildManifest.groovy

```diff
@@ -62,7 +62,7 @@ class BuildManifest implements Serializable {
             this.ref = data.ref
             this.commit_id = data.commit_id
             this.repository = data.repository
-            this.artifacts = new HashMap<>(data.artifacts)
+            this.artifacts = data.artifacts ? new HashMap<>(data.artifacts) : new HashMap<>()
```

Comment:
Could we add a test, or alter one of the existing tests to exercise this functionality?

Reply:
Note; I see that you altered the test data, but its not very clear how that alterative prevents the nullpointer, I was hoping you could create a new test that invokes `BuildManifest(...)` with artifacts undefined